src/qibolab/result.py

```diff
@@ -1,100 +1,91 @@
-from dataclasses import dataclass
 from functools import cached_property
 from typing import Optional
 
 import numpy as np
 import numpy.typing as npt
 
-ExecRes = np.dtype([("i", np.float64), ("q", np.float64)])
+iq_record_dtype = np.dtype([("i", np.float64), ("q", np.float64)])
 
 
-@dataclass
 class ExecutionResults:
     """Data structure to deal with the output of :func:`qibolab.platforms.abstract.AbstractPlatform.execute_pulse_sequence`"""
 
-    array: npt.NDArray[ExecRes]
-    shots: Optional[npt.NDArray[np.uint32]] = None
+    def __init__(self, i: np.ndarray | list, q: np.ndarray | list, state: np.ndarray | list = None):
+        if not isinstance(i, np.ndarray):
+            i = np.array(i)
+        if not isinstance(q, np.ndarray):
+            q = np.array(q)
+        if i.shape != q.shape:
+            raise ValueError("is_ and qs_ must have the same size")
 
-    @classmethod
-    def from_components(cls, is_, qs_, shots=None):
-        ar = np.empty(is_.shape, dtype=ExecRes)
-        ar["i"] = is_
-        ar["q"] = qs_
-        ar = np.rec.array(ar)
-        return cls(ar, shots)
+        self.voltage: npt.NDArray[iq_record_dtype] = np.recarray(i.shape, dtype=iq_record_dtype)
+        self.voltage["i"] = i
+        self.voltage["q"] = q
+
+        if state and not isinstance(state, np.ndarray):
+            state = np.array(state)
+        self.state: Optional[npt.NDArray[np.uint32]] = state
 
     @property
     def i(self):
-        return self.array.i
+        return self.voltage.i
 
     @property
     def q(self):
-        return self.array.q
-
-    def __add__(self, data):
-        assert len(data.i.shape) == len(data.q.shape)
-        if data.shots is not None:
-            assert len(data.i.shape) == len(data.shots.shape)
-        # concatenate on first dimension; if a scalar is passed, just append it
-        axis = 0 if len(data.i.shape) > 0 else None
-        i = np.append(self.i, data.i, axis=axis)
-        q = np.append(self.q, data.q, axis=axis)
-        if data.shots is not None:
-            shots = np.append(self.shots, data.shots, axis=axis)
-        else:
-            shots = None
-
-        new_execution_results = self.__class__.from_components(i, q, shots)
-
-        return new_execution_results
+        return self.voltage.q
 
     @cached_property
-    def measurement(self):
-        """Resonator signal voltage mesurement (MSR) in volts."""
+    def magnitude(self):
+        """Signal magnitude in volts."""
         return np.sqrt(self.i**2 + self.q**2)
 
     @cached_property
     def phase(self):
-        """Computes phase value."""
+        """Signal phase in radiants."""
```

Comment:
radians

Reply:
Seems good to me, but we either also add a .raw_wave  or create a very similar class for that (we would not use the probabilities and that would be it I think) so that we can accommodate all the possible results from the experiments.